orchestrator/graphql/resolvers/process.py

```diff
@@ -0,0 +1,231 @@
+        joinedload(ProcessTable.process_subscriptions)
+        .joinedload(ProcessSubscriptionTable.subscription)
+        .joinedload(SubscriptionTable.product),
+        defer("traceback"),
+    )
+
+    if _filter is not None:
+        if len(_filter) == 0 or (len(_filter) % 2) > 0:
+            raise_status(HTTPStatus.BAD_REQUEST, "Invalid number of filter arguments")
+        for filter_pair in chunked(_filter, 2):
+            field, value = filter_pair
+            field = field.lower()
+            if value is not None:
+                if field == "istask":
+                    value_as_bool = value.lower() in ("yes", "y", "ye", "true", "1", "ja")
+                    query = query.filter(ProcessTable.is_task.is_(value_as_bool))
+                elif field == "assignee":
+                    assignees = value.split("-")
+                    query = query.filter(ProcessTable.assignee.in_(assignees))
+                elif field == "status":
+                    statuses = value.split("-")
+                    query = query.filter(ProcessTable.last_status.in_(statuses))
+                elif field == "workflow":
+                    query = query.filter(ProcessTable.workflow.ilike("%" + value + "%"))
+                elif field == "creator":
+                    query = query.filter(ProcessTable.created_by.ilike("%" + value + "%"))
+                elif field == "organisation":
+                    try:
+                        value_as_uuid = UUID(value)
+                    except (ValueError, AttributeError):
+                        msg = "Not a valid customer_id, must be a UUID: '{value}'"
+                        logger.exception(msg)
+                        raise_status(HTTPStatus.BAD_REQUEST, msg)
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable)
+                        .filter(SubscriptionTable.customer_id == value_as_uuid)
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "product":
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable, ProductTable)
+                        .filter(ProductTable.name.ilike("%" + value + "%"))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "tag":
+                    tags = value.split("-")
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable, ProductTable)
+                        .filter(ProductTable.tag.in_(tags))
+                        .subquery()
+                    )
```

Comment:
We already join all required tables together at line 113, so we should be able to write this without a subquery. (Though I'll admit I don't know the required SQLAlchemy syntax by head)

Reply:
generic filtering is added in graphql setup PR.